Orange/widgets/visualize/owscatterplot.py

```diff
@@ -394,18 +395,40 @@ def set_subset_data(self, subset_data):
 
     # called when all signals are received, so the graph is updated only once
     def handleNewSignals(self):
-        self.graph.new_data(self.data_metas_X, self.subset_data)
+        if self.data is None or (self.data and not sp.issparse(self.data.X)):#  table.issparse() !!!
+            self.graph.new_data(self.data_metas_X, self.subset_data)
         if self.attribute_selection_list and \
                 all(attr in self.graph.domain
                         for attr in self.attribute_selection_list):
             self.attr_x = self.attribute_selection_list[0]
             self.attr_y = self.attribute_selection_list[1]
         self.attribute_selection_list = None
+        #if not sp.issparse(self.data.X):
         self.update_graph()
         self.cb_class_density.setEnabled(self.graph.can_draw_density())
         self.cb_reg_line.setEnabled(self.graph.can_draw_regresssion_line())
         self.apply_selection()
         self.unconditional_commit()
+        self.sparse_to_dense()
+
+    def sparse_to_dense(self):
+        if self.data is None or not self.data.is_sparse():
+            return
+        keys = []
+        for i, attr in enumerate(self.data.domain):
+            if attr in set([self.attr_x, self.attr_y, self.graph.attr_color]):
```

Comment:
I'm probably nitpicking here, but could you move this outside the loop into a variable so you do not create a new set in each iteration?